src/pages/blog/[slug].tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+import groq from 'groq'
+import {sanityClient} from 'utils/sanity-client'
+import imageUrlBuilder from '@sanity/image-url'
+import mdxComponents from 'components/mdx'
+import renderToString from 'next-mdx-remote/render-to-string'
```

Comment:
`next-mdx-remote` has limitations, but works very well for this use case. Primarily I believe you can't import so all components need to be pre-defined.

Reply:
If that bugs you, checkout mdx-bundler by yours truly 😉